lib/modules/manager/gradle/extract/consistent-versions-plugin.ts

```diff
@@ -0,0 +1,187 @@
+import { logger } from '../../../../logger';
+import { newlineRegex, regEx } from '../../../../util/regex';
+import type { PackageDependency } from '../../types';
+import type { GradleManagerData } from '../types';
+
+export const VERSIONS_PROPS = 'versions.props';
+export const VERSIONS_LOCK = 'versions.lock';
+
+/**
+ * Determines if Palantir Gradle consistent-versions is in use, https://github.com/palantir/gradle-consistent-versions.
+ * The plugin name must be in build file and both `versions.props` and `versions.lock` must exist.
+ *
+ * @param availableFiles list of gradle build files found in project
+ * @param fileContents map with file contents of all files
+ */
+export function usesGcv(
+  availableFiles: string[],
+  fileContents: Record<string, string | null>
+): boolean {
+  if (
+    fileContents['build.gradle']?.includes(
```

Comment:
1) This only works if there is a `build.gradle` in the main folder of the cloned repository. What about projects in subfolders? The same with the Kotlin pendant below.

2) What if the plugin is applied in a file other than `build.gradle` or `build.gradle.kts`? E.g. you use `apply from` to include it from another file.

Reply:
Using the `getSiblingFileName()` trick 1) should be ok.

But your point 2) is more tricky.. The plugin may be defined in a wholly different file. I see two options

1. Skip check for plugin active, and instead use presence of `versions.props` and `versions.lock` as siblings as que to start parsing. Worst case we parse out nothing and wasted some cycles.
2. Or check sibling `build.gradle(.kts)` first, and if no match, look in every single file in `fileContents` for `com.palantir.consistent-versions`. It's a bit more heavy, but will only be done if the two files are present and the plugin is not activated in top-level.

Reply:
A third option is to look for this text https://github.com/palantir/gradle-consistent-versions/blob/develop/src/main/java/com/palantir/gradle/versions/ConflictSafeLockFile.java#L38 in first line of `versions.lock`, which would be lightweight and a strongish indication that the plugin is (or was) active.

Reply:
I pushed a commit that implements the 3rd solution here, eliminating the need for inspecting all gradle build-files. 🏖